common/workunit/workunit.cpp

```diff
@@ -1409,6 +1450,19 @@ class GraphScopeIterator : public CInterfaceOf<IConstWUScopeIterator>
                     state = SGraphFirstSubGraph;
                 break;
             }
+            case SChildGraphFirstEdge:
+            {
+                Owned<IPropertyTreeIterator> treeIter = treeIters.tos().query().getElements("att/graph/edge");
+                if (treeIter && treeIter->first())
+                {
+                    treeIter.setown(createSortedIterator(*treeIter, compareEdgeNode));
```

Comment:
Would the original treeIter from 1455 be destroyed, when .setown is called but the new treeIter object 9created by createSortedIterator) is still using the original treeIter.

Reply:
The order is well defined - the setown function is only called after the parameters have been fully evaluated.  That will have called link on treeIter.  If link wasn't called by createSortedIterator() then it would be destroyed.  See other examples in the same file.

Reply:
CPTArrayIterator calls iter.get().  I thought that .get() doesn't link but I see from the definition now that IPropertyTreeIterator .get() does link.  For my own information, why isn't this function called iter.getLink?

Reply:
General convention is that functions starting get... always link while ones starting query... don't. Of course there are lots of places where that convention is broken, and in particular Owned<> and Linked<> break it...